azurerm/internal/acceptance/testing.go

```diff
@@ -0,0 +1,94 @@
+package acceptance
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/hashicorp/go-azure-helpers/authentication"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+var AzureProvider *schema.Provider
+var SupportedProviders map[string]terraform.ResourceProvider
+
+func PreCheck(t *testing.T) {
+	variables := []string{
+		"ARM_CLIENT_ID",
+		"ARM_CLIENT_SECRET",
+		"ARM_SUBSCRIPTION_ID",
+		"ARM_TENANT_ID",
+		"ARM_TEST_LOCATION",
+		"ARM_TEST_LOCATION_ALT",
+		"ARM_TEST_LOCATION_ALT2",
+	}
+
+	for _, variable := range variables {
+		value := os.Getenv(variable)
+		if value == "" {
+			t.Fatalf("`%s` must be set for acceptance tests!", variable)
+		}
+	}
+}
+
+func Location() string {
+	return os.Getenv("ARM_TEST_LOCATION")
+}
+
+func AltLocation() string {
```

Comment:
Part of me wants to rename this to `LocationAlt` while we do this but thats my OCD talking

Reply:
The other option would be to make Location a struct returning Primary/Default, Secondary and Ternary?